Queue tests fail

OS: `Ubuntu 20.04`
Tarantool: `2.8.2-0-gfc96d10f5`

For now, all queue tests fail.

```
$ cd queue/
$
$ mkdir snap xlog
$
$ tarantoolctl rocks install queue 1.1.0
$
$ tarantool ./config.lua > tarantool.log 2>&1 &
[1] 988812
$
$ cat tarantool.log
Deprecated option snap_dir, please use memtx_dir instead
2021-12-02 08:51:16.580 [988046] main/103/config.lua C> Tarantool 2.8.2-0-gfc96d10f5
2021-12-02 08:51:16.580 [988046] main/103/config.lua C> log level 5
2021-12-02 08:51:16.580 [988046] main/103/config.lua I> wal/engine cleanup is paused
2021-12-02 08:51:16.580 [988046] main/103/config.lua I> mapping 268435456 bytes for memtx tuple arena...
2021-12-02 08:51:16.580 [988046] main/103/config.lua I> Actual slab_alloc_factor calculated on the basis of desired slab_alloc_factor = 1.044274
2021-12-02 08:51:16.580 [988046] main/103/config.lua I> mapping 134217728 bytes for vinyl tuple arena...
2021-12-02 08:51:16.582 [988046] main/103/config.lua I> instance uuid a227e131-f9eb-4e7f-8b80-a4536bca71f7
2021-12-02 08:51:16.582 [988046] main/103/config.lua I> tx_binary: stopped
2021-12-02 08:51:16.582 [988046] main/103/config.lua I> tx_binary: bound to 0.0.0.0:3013
2021-12-02 08:51:16.582 [988046] main/103/config.lua I> initializing an empty data directory
2021-12-02 08:51:16.592 [988046] main/103/config.lua I> assigned id 1 to replica a227e131-f9eb-4e7f-8b80-a4536bca71f7
2021-12-02 08:51:16.592 [988046] main/103/config.lua I> cluster uuid c32cf3a6-70b5-485c-b9e5-913ad7300151
2021-12-02 08:51:16.592 [988046] snapshot/101/main I> saving snapshot `snap/00000000000000000000.snap.inprogress'
2021-12-02 08:51:16.607 [988046] snapshot/101/main I> done
2021-12-02 08:51:16.607 [988046] main/103/config.lua I> ready to accept requests
2021-12-02 08:51:16.607 [988046] main/104/gc I> wal/engine cleanup is resumed
2021-12-02 08:51:16.607 [988046] main/103/config.lua I> set 'log_level' configuration option to 5
2021-12-02 08:51:16.607 [988046] main/105/checkpoint_daemon I> scheduled next checkpoint for Thu Dec  2 10:12:11 2021
2021-12-02 08:51:16.608 [988046] main/103/config.lua I> set 'listen' configuration option to "3013"
2021-12-02 08:51:16.608 [988046] main/103/config.lua I> set 'log_format' configuration option to "plain"
2021-12-02 08:51:16.618 [988046] main schema.lua:2498 E> ER_NO_SUCH_SPACE: Space '_queue_taken' does not exist
2021-12-02 08:51:16.618 [988046] main F> fatal error, exiting the event loop
```

According to [this](https://github.com/tarantool/queue/tree/7d05b6db5b0d596db4281d53b57811e156234a0d#fields-of-the-_queue_consumers-space), `_queue_taken` is deprecated. So let's try to use `_queue_taken_2`. Open the `config.lua` file and change `box.schema.user.grant('test', 'read,write', 'space', '_queue_taken')` to `box.schema.user.grant('test', 'read,write', 'space', '_queue_taken_2')`.

```
$ ps aux | awk '/tarantool/{print $2}' | head -1 | xargs kill
$
$ rm -rf snap xlog && mkdir snap xlog
$
$ tarantool ./config.lua > tarantool.log 2>&1 &
[1] 988893
$
$ cat tarantool.log
Deprecated option snap_dir, please use memtx_dir instead
2021-12-02 09:10:55.567 [988893] main/103/config.lua C> Tarantool 2.8.2-0-gfc96d10f5
2021-12-02 09:10:55.567 [988893] main/103/config.lua C> log level 5
2021-12-02 09:10:55.567 [988893] main/103/config.lua I> wal/engine cleanup is paused
2021-12-02 09:10:55.567 [988893] main/103/config.lua I> mapping 268435456 bytes for memtx tuple arena...
2021-12-02 09:10:55.567 [988893] main/103/config.lua I> Actual slab_alloc_factor calculated on the basis of desired slab_alloc_factor = 1.044274
2021-12-02 09:10:55.567 [988893] main/103/config.lua I> mapping 134217728 bytes for vinyl tuple arena...
2021-12-02 09:10:55.568 [988893] main/103/config.lua I> instance uuid 8fead1ef-bff9-4e30-a857-c47ce4833944
2021-12-02 09:10:55.568 [988893] main/103/config.lua I> tx_binary: stopped
2021-12-02 09:10:55.569 [988893] main/103/config.lua I> tx_binary: bound to 0.0.0.0:3013
2021-12-02 09:10:55.569 [988893] main/103/config.lua I> initializing an empty data directory
2021-12-02 09:10:55.579 [988893] main/103/config.lua I> assigned id 1 to replica 8fead1ef-bff9-4e30-a857-c47ce4833944
2021-12-02 09:10:55.579 [988893] main/103/config.lua I> cluster uuid 1f211e77-88c4-4a38-9875-b4105ee1f45c
2021-12-02 09:10:55.580 [988893] snapshot/101/main I> saving snapshot `snap/00000000000000000000.snap.inprogress'
2021-12-02 09:10:55.589 [988893] snapshot/101/main I> done
2021-12-02 09:10:55.590 [988893] main/103/config.lua I> ready to accept requests
2021-12-02 09:10:55.590 [988893] main/104/gc I> wal/engine cleanup is resumed
2021-12-02 09:10:55.590 [988893] main/103/config.lua I> set 'log_level' configuration option to 5
2021-12-02 09:10:55.590 [988893] main/105/checkpoint_daemon I> scheduled next checkpoint for Thu Dec  2 10:52:58 2021
2021-12-02 09:10:55.590 [988893] main/103/config.lua I> set 'listen' configuration option to "3013"
2021-12-02 09:10:55.590 [988893] main/103/config.lua I> set 'log_format' configuration option to "plain"
2021-12-02 09:10:55.599 [988893] main C> entering the event loop
$
$ go clean -testcache && go test -v
=== RUN   TestFifoQueue
--- FAIL: TestFifoQueue (0.00s)
    queue_test.go:40: Failed to create queue: Create access to space 'test_queue' is denied for user 'test' (0x2a)
=== RUN   TestFifoQueue_GetExist_Statistic
--- FAIL: TestFifoQueue_GetExist_Statistic (0.00s)
    queue_test.go:65: Failed to create queue: Create access to space 'test_queue' is denied for user 'test' (0x2a)
=== RUN   TestFifoQueue_Put
--- FAIL: TestFifoQueue_Put (0.00s)
    queue_test.go:116: Failed to create queue: Create access to space 'test_queue' is denied for user 'test' (0x2a)
=== RUN   TestFifoQueue_Take
--- FAIL: TestFifoQueue_Take (0.03s)
    queue_test.go:159: Failed to create queue: Create access to space 'test_queue' is denied for user 'test' (0x2a)
=== RUN   TestFifoQueue_TakeTyped
--- FAIL: TestFifoQueue_TakeTyped (0.00s)
    queue_test.go:255: Failed to create queue: Create access to space 'test_queue' is denied for user 'test' (0x2a)
=== RUN   TestFifoQueue_Peek
--- FAIL: TestFifoQueue_Peek (0.00s)
    queue_test.go:332: Failed to create queue: Create access to space 'test_queue' is denied for user 'test' (0x2a)
=== RUN   TestFifoQueue_Bury_Kick
--- FAIL: TestFifoQueue_Bury_Kick (0.00s)
    queue_test.go:391: Failed to create queue: Create access to space 'test_queue' is denied for user 'test' (0x2a)
=== RUN   TestFifoQueue_Delete
--- FAIL: TestFifoQueue_Delete (0.00s)
    queue_test.go:478: Failed to create queue: Create access to space 'test_queue' is denied for user 'test' (0x2a)
=== RUN   TestFifoQueue_Release
--- FAIL: TestFifoQueue_Release (0.00s)
    queue_test.go:553: Failed to create queue: Create access to space 'test_queue' is denied for user 'test' (0x2a)
=== RUN   TestTtlQueue
--- FAIL: TestTtlQueue (0.00s)
    queue_test.go:643: Failed to create queue: Create access to space 'test_queue' is denied for user 'test' (0x2a)
=== RUN   TestTtlQueue_Put
--- FAIL: TestTtlQueue_Put (0.00s)
    queue_test.go:700: Failed to create queue: Create access to space 'test_queue' is denied for user 'test' (0x2a)
=== RUN   TestUtube_Put
--- FAIL: TestUtube_Put (0.00s)
    queue_test.go:772: Failed to create queue: Create access to space 'test_utube' is denied for user 'test' (0x2a)
FAIL
exit status 1
FAIL	github.com/tarantool/go-tarantool/queue	0.063s
```

--- a/.github/workflows/reusable_testing.yml
+++ b/.github/workflows/reusable_testing.yml
@@ -45,15 +45,14 @@ jobs:
           go clean -testcache && go test -v
           kill $TNT_PID
 
-# TODO(ylobankov): Uncomment this when tarantool/go-tarantool#115 is resolved.
-#      - name: Run queue tests
-#        working-directory: ./queue
-#        run: |
-#          mkdir snap xlog
-#          tarantoolctl rocks install queue 1.1.0
-#          TNT_PID=$(tarantool ./config.lua > tarantool.log 2>&1 & echo $!)
-#          go clean -testcache && go test -v
-#          kill $TNT_PID
+      - name: Run queue tests
+        working-directory: ./queue
+        run: |
+          mkdir snap xlog
+          tarantoolctl rocks install queue 1.1.0
+          TNT_PID=$(tarantool ./config.lua > tarantool.log 2>&1 & echo $!)
+          go clean -testcache && go test -v
+          kill $TNT_PID
 
       - name: Run uuid tests
         working-directory: ./uuid

--- a/.github/workflows/testing.yml
+++ b/.github/workflows/testing.yml
@@ -46,15 +46,16 @@ jobs:
           go clean -testcache && go test -v
           kill $TNT_PID
 
-# TODO(ylobankov): Uncomment this when tarantool/go-tarantool#115 is resolved.
-#      - name: Run queue tests
-#        working-directory: ./queue
-#        run: |
-#          mkdir snap xlog
-#          tarantoolctl rocks install queue 1.1.0
-#          TNT_PID=$(tarantool ./config.lua > tarantool.log 2>&1 & echo $!)
-#          go clean -testcache && go test -v
-#          kill $TNT_PID
+      - name: Run queue tests
+        working-directory: ./queue
+        run: |
+          rm -rf snap
+          rm -rf xlog
+          mkdir snap xlog
+          tarantoolctl rocks install queue 1.1.0
+          TNT_PID=$(tarantool ./config.lua > tarantool.log 2>&1 & echo $!)
+          go clean -testcache && go test -v
+          kill $TNT_PID
 
       - name: Run uuid tests
         working-directory: ./uuid

--- a/queue/config.lua
+++ b/queue/config.lua
@@ -18,6 +18,12 @@ box.schema.func.create('queue.tube.test_queue:delete')
 box.schema.func.create('queue.tube.test_queue:release')
 box.schema.func.create('queue.tube.test_queue:bury')
 box.schema.func.create('queue.statistics')
+box.schema.user.grant('test', 'create', 'space')
+box.schema.user.grant('test', 'write', 'space', '_schema')
+box.schema.user.grant('test', 'write', 'space', '_space')
+box.schema.user.grant('test', 'read,write', 'space', '_space_sequence')
+box.schema.user.grant('test', 'write', 'space', '_index')
+box.schema.user.grant('test', 'read, write', 'space', '_queue_session_ids')
 box.schema.user.grant('test', 'execute', 'universe')
 box.schema.user.grant('test', 'read,write', 'space', '_queue')
 box.schema.user.grant('test', 'read,write', 'space', '_schema')
@@ -25,5 +31,17 @@ box.schema.user.grant('test', 'read,write', 'space', '_space')
 box.schema.user.grant('test', 'read,write', 'space', '_index')
 box.schema.user.grant('test', 'read,write', 'space', '_queue_consumers')
 box.schema.user.grant('test', 'read,write', 'space', '_priv')
-box.schema.user.grant('test', 'read,write', 'space', '_queue_taken')
+box.schema.user.grant('test', 'read,write', 'space', '_queue_taken_2')
+if box.space._trigger ~= nil then
+    box.schema.user.grant('test', 'read', 'space', '_trigger')
+end
+if box.space._fk_constraint ~= nil then
+    box.schema.user.grant('test', 'read', 'space', '_fk_constraint')
+end
+if box.space._ck_constraint ~= nil then
+    box.schema.user.grant('test', 'read', 'space', '_ck_constraint')
+end
+if box.space._func_index ~= nil then
+    box.schema.user.grant('test', 'read', 'space', '_func_index')
+end
 end)

--- a/queue/queue_test.go
+++ b/queue/queue_test.go
@@ -13,7 +13,7 @@ import (
 
 var server = "127.0.0.1:3013"
 var opts = Opts{
-	Timeout: 500 * time.Millisecond,
+	Timeout: 2500 * time.Millisecond,
 	User:    "test",
 	Pass:    "test",
 	//Concurrency: 32,

